### PR TITLE
Utility functions

### PR DESCRIPTION
**Pull request by mihail911**
Moved utils.py into HyperSearch and moved some utility functions into utils

Please review @scottpurdy @oxtopus

### CODE REVIEW

File: nupic/support/nupic-default.xml [hunk 1]
```diff
@@ -113,7 +113,7 @@
 
 <property>
   <name>nupic.hypersearch.speculative.particles.default</name>
-  <value>1</value>
```
Comment (scottpurdy) [inline nupic/support/nupic-default.xml]: Don't submit this change.

**Comment by scottpurdy** [comment]
Make "HyperSearch" directory all lowercase: `nupic.swarming.hypersearch`

File: nupic/swarming/HyperSearch/__init__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
```
Comment (scottpurdy) [inline nupic/swarming/HyperSearch/__init__.py]: file must end with single newline character

Comment (scottpurdy) [inline nupic/swarming/HyperSearch/__init__.py]: Make sure you run pylint on files before submitting them

File: nupic/swarming/HypersearchV2.py [hunk 3]
```diff
@@ -36,13 +36,13 @@
...
-from nupic.support import clippedObj
+from nupic.swarming.HyperSearch.utils import clippedObj
 from nupic.support.serializationutils import sortedJSONDumpS
 from nupic.support.configuration import Configuration
 from nupic.support.errorcodes import ErrorCodes
 from nupic.database.ClientJobsDAO import (
     ClientJobsDAO, InvalidConnectionException)
-from nupic.swarming.utils import (runModelGivenBaseAndParams,
+from nupic.swarming.HyperSearch.utils import (runModelGivenBaseAndParams,
                                                   runDummyModel)
```
Comment (scottpurdy) [inline nupic/swarming/HypersearchV2.py]: fix indentation on wrapped line

**Comment by scottpurdy** [comment]
done with review. Nothing major

**Comment by oxtopus** [comment]
Why `nupic.swarming.hypersearch` and not `nupic.swarming` or `nupic.hypersearch`?  `swarming` and `hypersearch` are synonymous.

**Comment by scottpurdy** [comment]
@oxtopus - The idea is that `nupic.swarming` will be the NuPIC client of the Hypersearch library. We want to pull the Hypersearch library out in pieces, so we will move it into a temporary location `nupic.swarming.hypersearch` which we will later move out into standalone library. Does that seem like a good strategy to you?

**Comment by mihail911** [comment]
Ran pylint and made requested changes. @scottpurdy

**Comment by scottpurdy** [comment]
there is still the data file

**Comment by mihail911** [comment]
@scottpurdy

**Comment by scottpurdy** [comment]
Please revert first_order_0.csv. Otherwise, looks good :+1:

**Comment by mihail911** [comment]
Reverted. @scottpurdy

**Comment by scottpurdy** [comment]
:+1: once build passes

**Comment by mihail911** [comment]
Migrating these changes to cleaner PR

**Comment by scottpurdy** [comment]
@mihail911 - I don't like closing PRs and opening new ones because you lose the conversation history. Please try to avoid in the future.

**Comment by cogmission** [comment]
Omg! We should submit this PR to the Guinness Book of World Records! :-)

I've never seen so many commits under one PR!

LOL!

Gread work though @mihail911 !

**Comment by oxtopus** [comment]
Um.  yeah.  It looks like an interactive rebase gone awry.

**Comment by cogmission** [comment]
@oxtopus lmao!

**Comment by cogmission** [comment]
unknown

**Comment by EronWright** [comment]
This PR shall be the poster child for 'git squash'.

**Comment by mihail911** [comment]
I'd like to thank the Academy, my parents, and my manager Scott for inspiring me everyday.

**Comment by akhilaananthram** [comment]
:clap:

**Comment by cogmission** [comment]
:-P

**Comment by passiweinberger** [comment]
Omg :D

**Pull request closed without merging**